src/transformers/models/ict/configuration_ict.py

```diff
@@ -0,0 +1,299 @@
+# coding=utf-8
+# Copyright 2023 Authors at City University of Hong Kong, Microsoft Cloud + AI,
+# The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" ICT model configuration"""
+
+
+
+import copy
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+ICT_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "sheonhan/ict-imagenet-32": "https://huggingface.co/sheonhan/ict-imagenet-32/resolve/main/config.json",
+}
+
+
+class ICTGuidedUpsamplerConfig(PretrainedConfig):
```

Comment:
So I separated the configs to two different classes like [Align config](https://github.com/alaradirik/transformers/blob/main/src/transformers/models/align/configuration_align.py) as suggested by @alaradirik

Reply:
Looks pretty neat and comprehensive to me :)